LoopKitUI/Views/Settings Editors/CorrectionRangeScheduleEditor.swift

```diff
@@ -40,14 +42,13 @@ public struct CorrectionRangeScheduleEditor: View {
            viewModel: TherapySettingsViewModel,
            didSave: (() -> Void)? = nil
     ) {
-        precondition(viewModel.therapySettings.glucoseUnit != nil)
         self.init(
             schedule: viewModel.therapySettings.glucoseTargetRangeSchedule,
-            unit: viewModel.therapySettings.glucoseUnit!,
+            unit: viewModel.therapySettings.glucoseTargetRangeSchedule?.unit ?? viewModel.preferredGlucoseUnit,
             minValue: Guardrail.minCorrectionRangeValue(suspendThreshold: viewModel.therapySettings.suspendThreshold),
             onSave: { [weak viewModel] newSchedule in
-                viewModel?.saveCorrectionRange(range: newSchedule)
                 didSave?()
+                viewModel?.saveCorrectionRange(range: newSchedule)
```

Comment:
I don't think this is correct. `didSave` is intentionally called _after_ the viewModel saves the data.

Reply:
Ah, this was leftover from experiments of trying to fix the non-dismissing schedules in iOS 14 (didSave was the dismiss call). I reverted it, and my other fix (dispatch async) still works. Will revert this.